assets/shared/helpers/player/videopress-adapter.js

```diff
@@ -61,7 +74,25 @@ export const getDuration = ( player ) =>
 			reject( new Error( 'Video duration not found' ) );
 		}
 
-		resolve( parseFloat( player.dataset.duration ) );
+		resolve( parseFloat( duration ) );
+	} );
+
+/**
+ * Get the current video time.
+ *
+ * @param {HTMLIFrameElement} player The player element.
+ *
+ * @return {Promise<number>} The current video time in seconds through a promise.
+ */
+export const getCurrentTime = ( player ) =>
+	new Promise( ( resolve, reject ) => {
+		const { currentTime } = player.dataset;
+
+		if ( ! currentTime ) {
+			reject( new Error( 'Video current time not found' ) );
+		}
+
+		resolve( parseFloat( player.dataset.currentTime ) );
```

Comment:
Maybe we could use `currentTime` here directly?

```suggestion
		resolve( parseFloat( currentTime ) );
```